src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/RespondNotificationService.java

```diff
@@ -32,11 +128,145 @@ public List<DynamicValueType> getSendNotificationSelection(CaseData caseData) {
         });
     }
 
+    private String getRespondNotificationSingleDocumentMarkdown(UploadedDocumentType uploadedDocumentType) {
+        String document = "| Supporting material | |";
+        if (uploadedDocumentType != null) {
+            document = String.format("| Supporting material | %s", createLinkForUploadedDocument(uploadedDocumentType));
+        }
+        return document;
+    }
+
+    private String getRespondNotificationDocumentsMarkdown(RespondNotificationType respondNotificationType) {
+        if (respondNotificationType.getRespondNotificationUploadDocument() == null) {
+            return "| Supporting material | |";
+        }
+        List<String> documents = respondNotificationType.getRespondNotificationUploadDocument().stream()
+            .map(documentTypeItem ->
+                getRespondNotificationSingleDocumentMarkdown(documentTypeItem.getValue().getUploadedDocument()))
+            .collect(Collectors.toList());
+        return String.join("\r\n", documents);
+    }
+
+    /**
+     * Gets markdown containing the details of the notification responses.
+     * @param sendNotificationType notification
+     * @return markdown of responses
+     */
+    public String getRespondNotificationMarkdown(SendNotificationType sendNotificationType) {
+        var respondNotificationTypeCollection = sendNotificationType.getRespondNotificationTypeCollection();
+        if (respondNotificationTypeCollection == null) {
+            return "";
+        }
+        List<String> respondNotificationMarkdownList =
+            respondNotificationTypeCollection.stream().map(respondNotificationType -> {
+                Integer index =
+                    respondNotificationTypeCollection.indexOf(respondNotificationType) + 1;
```

Comment:
fori loop might be nicer

Reply:
In this case, I think the stream is fine